src/UXClient/Models/ChartOptions.ts

```diff
@@ -73,7 +73,7 @@ class ChartOptions {
     public shouldSticky: boolean; // whether sticky is triggered in the linechart when stickySeries or unStickySeries is called
     public strings: any; // passed in key value pairs of strings -> strings
     public suppressResizeListener: boolean; // whether a component's resize function is ignored. Applies to components which draw an SVG
-    public swimLaneOptions: any;  // mapping of swim lane number to information about that swimlane, including axis type
+    public swimLaneOptions: {[key: number]: {yAxisType?: YAxisStates, label?: string, onClick?: (lane: number) => any, collapseEvents?: string}} | null;  // mapping of swim lane number to information about that swimlane, including axis type
```

Comment:
do you think making all of the props in value of the object optional is good practice? like would there be a case like swimLaneOptions: {1: {}} better than swimLaneOptions: null?
And, maybe we can define interface for the value of {yAxisType?: YAxisStates, label?: string, onClick?: (lane: number) => any, collapseEvents?: string} or whole key-value pair
I don't know, feels like something can be improved here :)

Reply:
So I was wondering that too.  I think one of the problems is, because the render function of chart components takes `options: any`, you don't actually see these typings anyway.  That's why so many of the objects here are type `any`.

 I know that label, onClick, and collapseEvents are optional.  I'm not sure about yAxisType though.  @niusoff do you have any thoughts on these typings for swimLaneOptions?

@iremgokce, so do you mean to define an interface for these types outside of chartOptions?  And what do you mean by key-value pair?

Reply:
Talked to Nick.  Making yAxisType non-optional.  I'll look into moving this typing to an interface.